squad/core/tasks/__init__.py

```diff
@@ -262,6 +279,67 @@ def get_suite_version(test_run, suite):
         return None
 
 
+def save_delayed_report(delayed_report, error_message, status_code):
```

Comment:
The method name is a bit confusing here. It updates two specific field in the report, so probably go in that direction.